django/contrib/admin/filters.py

```diff
@@ -420,3 +420,47 @@ class RelatedOnlyFieldListFilter(RelatedFieldListFilter):
     def field_choices(self, field, request, model_admin):
         pk_qs = model_admin.get_queryset(request).distinct().values_list('%s__pk' % self.field_path, flat=True)
         return field.get_choices(include_blank=False, limit_choices_to={'pk__in': pk_qs})
+
+
+class EmptyFieldListFilter(FieldListFilter):
+    def __init__(self, field, request, params, model, model_admin, field_path):
+        self.lookup_kwarg = '%s__isempty' % field_path
+        self.lookup_val = params.get(self.lookup_kwarg)
+        super().__init__(field, request, params, model, model_admin, field_path)
+
+    def queryset(self, request, queryset):
+        if self.lookup_kwarg not in self.used_parameters:
+            return queryset
+
+        if self.lookup_val not in ('0', '1'):
+            raise IncorrectLookupParameters(
+                'Incorrect lookup value %s provided for parameter %s'
+                % (self.lookup_val, self.lookup_kwarg)
+            )
+
+        query_filter = {}
+
+        if self.field.empty_strings_allowed:
+            query_filter[self.field_path] = ''
+        if self.field.null:  # Allow null value
+            query_filter['%s__isnull' % self.field_path] = True
+
+        lookup_condition = models.Q(**query_filter, _connector=models.Q.OR)
```

Comment:
Avoid using the private `_connector` API. The following should work just fine.

```python
lookup_condition = Q()
if self.field.empty_strings_allowed:
    lookup_condition |= Q(**{self.field_path: ''})
if self.field.null:
    lookup_condition |= Q(**{'%s__isnull' % self.field_path: True})

Reply:
I didn't know about this, great tip thanks :+1: